File: Data Structures & Algorithms/binary-tree-diameter/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private int ans = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        if(root == null) return 0;
        System.out.println("root : "+root+ " : "+root.val);
        ans = Math.max(maxDepth(root.left)+maxDepth(root.right),ans);
        System.out.println("ans : "+ans);
        diameterOfBinaryTree(root.left);
        diameterOfBinaryTree(root.right);
        return ans; 
    }

    public int maxDepth(TreeNode root){
        if(root == null) return 0;

        return Math.max(maxDepth(root.left)+1,maxDepth(root.right)+1);
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */

    isBalanced(root) {
        if(root === null ) return true;
        if(this.isBalanced(root.left) === false) return false;
        if(this.isBalanced(root.right)=== false) return false;

        if(Math.abs(this.maxDepth(root.left)-this.maxDepth(root.right))>1){
            return false;
        } 
        return true;
    }

    maxDepth(root){
        if(root === null) return 0;

        return Math.max(this.maxDepth(root.left),this.maxDepth(root.right))+1;

    }
}
